test/PackageVersionsTest/InstallerTest.php

```diff
@@ -142,6 +142,61 @@ public function testDumpVersionsClassIfExistingFileIsNotWritable() : void
         $this->rmDir($vendorDir);
     }
 
+    public function testDumpVersionsClassIfReadonlyFilesystem() : void
+    {
+        $config            = $this->createMock(Config::class);
+        $locker            = $this->createMock(Locker::class);
+        $repositoryManager = $this->createMock(RepositoryManager::class);
+        $installManager    = $this->createMock(InstallationManager::class);
+        $repository        = $this->createMock(InstalledRepositoryInterface::class);
+
+        $vendorDir = sys_get_temp_dir() . '/' . uniqid('InstallerTest', true);
+
+        $expectedPath = $vendorDir . '/ocramius/package-versions/src/PackageVersions';
+
+        /** @noinspection MkdirRaceConditionInspection */
+        mkdir($expectedPath, 0700, true);
+
+        $expectedFileName = $expectedPath . '/Versions.php';
+        file_put_contents($expectedFileName, 'NOT PHP!');
+        chmod($expectedFileName, 0400);
+        chmod($expectedPath, 0400);
+
+        $locker
+            ->method('getLockData')
+            ->willReturn([
+                'packages' => [
+                    [
+                        'name'    => 'ocramius/package-versions',
+                        'version' => '1.0.0',
+                    ],
+                ],
+            ]);
+
+        $repositoryManager->method('getLocalRepository')->willReturn($repository);
+
+        $this->composer->method('getConfig')->willReturn($config);
+        $this->composer->method('getLocker')->willReturn($locker);
+        $this->composer->method('getRepositoryManager')->willReturn($repositoryManager);
+        $this->composer->method('getPackage')->willReturn($this->getRootPackageMock());
+        $this->composer->method('getInstallationManager')->willReturn($installManager);
+
+        $config->method('get')->with('vendor-dir')->willReturn($vendorDir);
+
+        Installer::dumpVersionsClass(new Event(
+            'post-install-cmd',
+            $this->composer,
+            $this->io
+        ));
+
+        chmod($expectedPath, 0700);
```

Comment:
Can we move the altering of the file to **after** the assertions? Ideally, I'd expect this to happen in `rmDir`

Reply:
For some reason `file_get_contents` not working without this. (logically, it should work, because file is readable)
![image](https://user-images.githubusercontent.com/9398439/85201478-07860780-b319-11ea-94a6-fc91aaefc831.png)

Reply:
Let's keep it then :+1: